exporters/committime/collector_image.py

```diff
@@ -121,9 +121,15 @@ def _set_commit_timestamp(self, metric: CommitMetric, errors) -> CommitMetric:
         # Only convert when commit_time is in metric, previously should be
         # found from the Label with fallback to annotation
         if metric.commit_time:
-            metric.commit_timestamp = parse_guessing_timezone_DYNAMIC(
-                metric.commit_time, format=self._timedate_format
-            ).timestamp()
+            try:
+                metric.commit_timestamp = to_epoch_from_string(
+                    metric.commit_time
+                ).timestamp()
+            except (ValueError, AttributeError):
```

Comment:
When would an AttributeError happen?

Reply:
When you pass integer and not str(int) as input for example.

Reply:
Shouldn't we check for that and raise a `TypeError` ourselves, so we know where it came from?

Reply:
Now that I think about it-- when would `metric.commit_time` ever be anything but a string?